wp-includes/class-wp-service-workers.php

```diff
@@ -83,6 +187,185 @@ public function register( $handle, $src, $deps = array(), $scope = self::SCOPE_A
+	 *                         WP_Service_Workers::STRATEGY_NETWORK_FIRST, WP_Service_Workers::STRATEGY_CACHE_ONLY,
+	 *                         WP_Service_Workers::STRATEGY_NETWORK_ONLY.
+	 * @param array  $strategy_args {
+	 *     An array of strategy arguments.
+	 *
+	 *     @type string $cache_name Cache name. Optional.
+	 *     @type array  $plugins    Array of plugins with configuration. The key of each plugin in the array must match the plugin's name.
+	 *                              See https://developers.google.com/web/tools/workbox/guides/using-plugins#workbox_plugins.
+	 * }
+	 */
+	public function register_cached_route( $route, $strategy, $strategy_args = array() ) {
+
+		if ( ! in_array( $strategy, array(
+			self::STRATEGY_STALE_WHILE_REVALIDATE,
+			self::STRATEGY_CACHE_FIRST,
+			self::STRATEGY_CACHE_ONLY,
+			self::STRATEGY_NETWORK_FIRST,
+			self::STRATEGY_NETWORK_ONLY,
+		), true ) ) {
+			_doing_it_wrong( __METHOD__, esc_html__( 'Strategy must be either WP_Service_Workers::STRATEGY_STALE_WHILE_REVALIDATE, WP_Service_Workers::STRATEGY_CACHE_FIRST,
+	            WP_Service_Workers::STRATEGY_NETWORK_FIRST, WP_Service_Workers::STRATEGY_CACHE_ONLY, or WP_Service_Workers::STRATEGY_NETWORK_ONLY.', 'pwa' ), '0.2' );
+			return;
+		}
+
+		if ( ! is_string( $route ) ) {
+			/* translators: %s is caching strategy */
+			$error = sprintf( __( 'Route for the caching strategy %s must be a string.', 'pwa' ), $strategy );
+			_doing_it_wrong( __METHOD__, esc_html( $error ), '0.2' );
+		} else {
+
+			$this->registered_caching_routes[] = array(
+				'route'         => $route,
+				'strategy'      => $strategy,
+				'strategy_args' => $strategy_args,
+			);
+		}
+	}
+
+	/**
+	 * Register routes / files for precaching.
+	 *
+	 * @param array $routes {
+	 *      Array of routes.
+	 *
+	 *      @type string $url      URL of the route.
+	 *      @type string $revision Revision (optional).
+	 * }
+	 */
+	public function register_precached_routes( $routes ) {
+		if ( ! is_array( $routes ) || empty( $routes ) ) {
+			_doing_it_wrong( __METHOD__, esc_html__( 'Routes must be an array.', 'pwa' ), '0.2' );
+			return;
+		}
+		$this->registered_precaching_routes = array_merge(
+			$routes,
+			$this->registered_precaching_routes
+		);
+	}
+
+	/**
+	 * Gets the script for precaching routes.
+	 *
+	 * @param array $routes Array of routes.
+	 * @return string Precaching logic.
+	 */
+	protected function get_precaching_for_routes_script( $routes ) {
+
+		$routes_list = array();
+		foreach ( $routes as $route ) {
+			if ( is_string( $route ) ) {
+				$route = array( 'url' => $route );
+			}
+			if ( ! isset( $route['revision'] ) ) {
+				$route['revision'] = get_bloginfo( 'version' );
+			}
+
+			$routes_list[] = $route;
+		}
+		if ( empty( $routes_list ) ) {
+			return '';
+		}
+		return sprintf( "wp.serviceWorker.precaching.precacheAndRoute( %s );\n", wp_json_encode( $routes_list ) );
```

Comment:
Should the `precacheAndRoute` call here include the `directoryIndex: null` option? In a WordPress context, it doesn't make sense that Workbox would look for `index.html`.

https://developers.google.com/web/tools/workbox/modules/workbox-precaching#directory_index

The same goes for the `cleanUrls` option. I think it should be disabled too.